extensions/amp-story/1.0/amp-story.js

```diff
@@ -2634,9 +2634,57 @@ export class AmpStory extends AMP.BaseElement {
       this.next_();
     } else if (data['previous']) {
       this.previous_();
+    } else if (data['delta']) {
+      this.switchDelta_(data['delta']);
     }
   }
 
+  /**
+   * Switches to a page in the story given a delta. If new index is out of
+   * bounds, it will wrap around the pages array.
+   * @param {number} delta
+   * @private
+   */
+  switchDelta_(delta) {
+    const activePage = devAssert(
+      this.activePage_,
+      'No active page set when navigating to pageId.'
+    );
+
+    const currentPageIdx = findIndex(
+      this.pages_,
+      (page) => page.element.id === activePage.element.id
+    );
+
+    const newPageIdx = currentPageIdx + delta;
+    const targetPage =
+      delta > 0
+        ? this.pages_[newPageIdx % this.pages_.length]
```

Comment:
So if there are two pages left but I send `go(0, 3)`, it rewinds the story?
I think we should stick to using `go()` like the `history` API, where `go(0, 99999)` would just go to the last page and `go(0, -99999)` to the first one.

Not sure about what should happen when we get a `go(0, 1)` when on the last page. If we want people to build page navigation buttons, the expected behavior would be to navigate to the next story. We could do that in another iteration, but rewinding the story is definitely not what we should do.

Reply:
Agree with this, was on the edge but it makes more sense to not wrap 👍